### DISCUSSION TITLE
two-factor authentication

### DISCUSSION THREAD

**Discussion by ghost** [community/community#144085] [Discussions]
### Select Topic Area

General

### Body

Hello

I receive the message I need to enable two-factor authentication before December 23.
What happens if I don't do that. Can I still make repositories and upload source code?

**Comment by dalvarez2596** [comment]
Hi @openrct2girl, probably you wont be ale to login until you enable your two factor authentication according to this :
<img width="755" alt="Screenshot 2024-11-09 at 1 36 43 PM" src="https://github.com/user-attachments/assets/6663dde4-047a-43c2-a0ab-e78d5763891d">
So even if you not setup right now, you will have to in the future, in order to access your account.
You can refer to this [About 2FA](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/about-mandatory-two-factor-authentication) too see more info.

**Reply by ghost** [reply]
Is there an alternative to github that does not require 2FA? I don't like this

**Reply by dalvarez2596** [reply]
Well, there are other tools for version control based on Git, like Bitbucket (I think you don’t need to use 2FA unless you're part of an organization and the admin requires it). I think for GitLab, it’s mandatory as well. I was like you—I didn’t like the idea at first, but now I’ve gotten used to it. It’s a security measure for your account, and most apps now use 2FA to protect user accounts.

**Comment by Ravindulakmina** [comment]
Hello!
--
**You will have no problem doing it**

Here’s what will happen if you don’t enable 2FA:

1. **Restricted Access**: GitHub may prevent you from accessing certain features, such as creating or contributing to repositories. You could be locked out of activities like pushing code, creating pull requests, and managing repositories until 2FA is enabled.
   
2. **Repository Management**: You might still be able to view repositories and interact with some read-only features. However, actions that involve contributing or modifying code would be restricted.

3. **Potential Account Lockout**: Over time, not enabling 2FA could even result in account suspension, limiting your access further.

To avoid these restrictions, it’s best to set up 2FA as soon as possible. You can enable it under **Settings > Account security** in your GitHub profile. You’ll have the option to use a mobile app, SMS, or a security key for authentication.

If you need help with 2FA setup, GitHub’s [2FA guide](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa) provides a step-by-step process.

Enabling 2FA will secure your account and allow uninterrupted access to GitHub’s full functionality. Let me know if you need more guidance!

**Please mark as accepted answer if this helped you**
